### PR TITLE
csi: skip creating networkFence when csi disabled

### LINKED ISSUE #14223: Dont create the Fencing CR when CSI is disabled
Today we check for the CRD and create the Fencing CR when a annotation is added on a node, we need to skip this is CSI is disabled.

### PR DESCRIPTION
**Pull request by subhamkrai**
skip creating networkFence when 'ROOK_CSI_DISABLE_DRIVER' is set to 'false'.

**Issue resolved by this Pull Request:**
Resolves #14223 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
will mark this to ready when testing is done

File: pkg/operator/ceph/cluster/watcher.go [hunk 1]
```diff
@@ -168,6 +168,16 @@ func (c *clientCluster) handleNodeFailure(ctx context.Context, cluster *cephv1.C
 		return nil
 	}
 
+	disabledCSI, err := k8sutil.GetOperatorSetting(ctx, c.context.Clientset, opcontroller.OperatorSettingConfigMapName, "ROOK_CSI_DISABLE_DRIVER", "false")
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to get configmap value `ROOK_WATCH_FOR_NODE_FAILURE`.")
```
Comment (BlaineEXE) [inline pkg/operator/ceph/cluster/watcher.go:173]:
```suggestion
		return pkgerror.Wrapf(err, "failed to get configmap value `ROOK_CSI_DISABLE_DRIVER`.")
```

Comment (subhamkrai) [reply pkg/operator/ceph/cluster/watcher.go]: copy-paste mistake :XD

**Comment by BlaineEXE** [comment]
This might need to be rebased after https://github.com/rook/rook/pull/14299

File: pkg/operator/ceph/cluster/watcher.go [hunk 2]
```diff
@@ -168,6 +168,16 @@ func (c *clientCluster) handleNodeFailure(ctx context.Context, cluster *cephv1.C
 		return nil
 	}
 
+	disabledCSI, err := k8sutil.GetOperatorSetting(ctx, c.context.Clientset, opcontroller.OperatorSettingConfigMapName, "ROOK_CSI_DISABLE_DRIVER", "false")
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to get configmap value `ROOK_CSI_DISABLE_DRIVER`.")
+	}
+
+	if strings.ToLower(disabledCSI) != "false" {
+		logger.Debugf("not watching for node failures since `ROOK_CSI_DISABLE_DRIVER` is set to %q, skip creating networkFence", watchForNodeLoss)
```
Comment (Madhu-1) [inline pkg/operator/ceph/cluster/watcher.go:177]: need to use `disabledCSI` for logging

Comment (subhamkrai) [reply pkg/operator/ceph/cluster/watcher.go]: ack

**Review by travisn — COMMENTED** [review]
@subhamkrai Is this ready for merge or still being tested since it's in Draft state?

**Comment by subhamkrai** [comment]
> @subhamkrai Is this ready for merge or still being tested since it's in Draft state?

forgot to marked it ready for review, testing looks good.

```
2024-06-25 11:18:18.536001 D | ceph-cluster-controller: not watching for node failures since `ROOK_CSI_DISABLE_DRIVER` is set to "true", skip creating networkFence
```

```
$ kubectl taint nodes ip-10-0-83-49.us-east-2.compute.internal node.kubernetes.io/out-of-service=nodeshutdown:NoExecute
node/ip-10-0-83-49.us-east-2.compute.internal tainted
~/go/src/github.com/rook/deploy/examples
srai@fedora ~ (skip-fence-when-csi-disabled) $ kc get pods
NAME                                                              READY   STATUS      RESTARTS   AGE
rook-ceph-crashcollector-42212ea233898be63885845407eeda5e-5q8z6   1/1     Running     0          4m4s
rook-ceph-crashcollector-63b70ec0d037fadd38c0c78e8900a63c-6lstp   1/1     Running     0          4m31s
rook-ceph-exporter-ip-10-0-26-200.us-east-2.compute.intern84gff   1/1     Running     0          4m4s
rook-ceph-exporter-ip-10-0-90-216.us-east-2.compute.internljnz2   1/1     Running     0          4m31s
rook-ceph-mgr-a-6898bbcdf8-c4p2s                                  3/3     Running     0          4m31s
rook-ceph-mgr-b-5df76ddb57-n7wpk                                  3/3     Running     0          2m1s
rook-ceph-mon-a-5f86854944-q8hqf                                  2/2     Running     0          6m12s
rook-ceph-mon-b-6465d8766b-pwdfx                                  0/2     Pending     0          2m1s
rook-ceph-mon-c-6bcc499d4d-lck8l                                  2/2     Running     0          5m4s
rook-ceph-operator-b9f78dcb6-4t84v                                1/1     Running     0          7m28s
rook-ceph-osd-0-75f647784b-vtt9p                                  2/2     Running     0          3m41s
rook-ceph-osd-1-5bdc7ffc86-dnnjj                                  2/2     Running     0          2m1s
rook-ceph-osd-2-776d9695-sclt6                                    2/2     Running     0          3m22s
rook-ceph-osd-prepare-set1-data-0jmmxj-nlrdd                      0/1     Completed   0          3m56s
rook-ceph-osd-prepare-set1-data-1g8gq6-dcx97                      0/1     Completed   0          3m56s
rook-ceph-osd-prepare-set1-data-2985v6-76gm5                      0/1     Completed   0          3m56s
```

**Review by Madhu-1 — APPROVED** [review]

**Review by BlaineEXE — APPROVED** [review]

**Review by parth-gr — APPROVED** [review]

**Comment by BlaineEXE** [comment]
@subhamkrai does this need backported to 1.14?

**Pull request merged**